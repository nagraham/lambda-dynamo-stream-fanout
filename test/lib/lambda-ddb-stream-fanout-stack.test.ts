import { SynthUtils } from '@aws-cdk/assert';
import * as core from '@aws-cdk/core';
import LambdaDdbStreamFanout = require('../../lib/lambda-ddb-stream-fanout-stack');

test('SNAPSHOT: Lambda dynamodb fanout stack', () => {
  const app = new core.App();
  const stack = new LambdaDdbStreamFanout.LambdaDdbStreamFanoutStack(app, 'TestFanoutStack');
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
})

// test('Empty Stack', () => {
//     const app = new cdk.App();
//     // WHEN
//     const stack = new LambdaDdbStreamFanout.LambdaDdbStreamFanoutStack(app, 'MyTestStack');
//     // THEN
//     expectCDK(stack).to(matchTemplate({
//       "Resources": {}
//     }, MatchStyle.EXACT))
// });
