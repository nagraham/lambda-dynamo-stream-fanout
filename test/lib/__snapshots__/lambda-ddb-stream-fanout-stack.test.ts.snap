// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOT: Lambda dynamodb fanout stack 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters2fd382a2d561f34ad0a0f69f0ca93deb0a54e0cd06d88f4165ff6508f79def65ArtifactHashFC2D96DA": Object {
      "Description": "Artifact hash for asset \\"2fd382a2d561f34ad0a0f69f0ca93deb0a54e0cd06d88f4165ff6508f79def65\\"",
      "Type": "String",
    },
    "AssetParameters2fd382a2d561f34ad0a0f69f0ca93deb0a54e0cd06d88f4165ff6508f79def65S3BucketD4DF0C4F": Object {
      "Description": "S3 bucket for asset \\"2fd382a2d561f34ad0a0f69f0ca93deb0a54e0cd06d88f4165ff6508f79def65\\"",
      "Type": "String",
    },
    "AssetParameters2fd382a2d561f34ad0a0f69f0ca93deb0a54e0cd06d88f4165ff6508f79def65S3VersionKey0455DD78": Object {
      "Description": "S3 key for asset version \\"2fd382a2d561f34ad0a0f69f0ca93deb0a54e0cd06d88f4165ff6508f79def65\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "LogInputLambda9AB18616": Object {
      "DependsOn": Array [
        "LogInputLambdaServiceRoleDefaultPolicyC216A535",
        "LogInputLambdaServiceRoleCA6E9A50",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters2fd382a2d561f34ad0a0f69f0ca93deb0a54e0cd06d88f4165ff6508f79def65S3BucketD4DF0C4F",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters2fd382a2d561f34ad0a0f69f0ca93deb0a54e0cd06d88f4165ff6508f79def65S3VersionKey0455DD78",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters2fd382a2d561f34ad0a0f69f0ca93deb0a54e0cd06d88f4165ff6508f79def65S3VersionKey0455DD78",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "handlers.logInput",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LogInputLambdaServiceRoleCA6E9A50",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogInputLambdaServiceRoleCA6E9A50": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogInputLambdaServiceRoleDefaultPolicyC216A535": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "dynamodb:ListStreams",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "TodoTable20200118D40ADDDF",
                        "Arn",
                      ],
                    },
                    "/stream/*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "dynamodb:DescribeStream",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "TodoTable20200118D40ADDDF",
                  "StreamArn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogInputLambdaServiceRoleDefaultPolicyC216A535",
        "Roles": Array [
          Object {
            "Ref": "LogInputLambdaServiceRoleCA6E9A50",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TodoLambda824170AD": Object {
      "DependsOn": Array [
        "TodoLambdaServiceRoleDefaultPolicy7E5F178B",
        "TodoLambdaServiceRoleB5E6BE45",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters2fd382a2d561f34ad0a0f69f0ca93deb0a54e0cd06d88f4165ff6508f79def65S3BucketD4DF0C4F",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters2fd382a2d561f34ad0a0f69f0ca93deb0a54e0cd06d88f4165ff6508f79def65S3VersionKey0455DD78",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters2fd382a2d561f34ad0a0f69f0ca93deb0a54e0cd06d88f4165ff6508f79def65S3VersionKey0455DD78",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "handlers.handleTodos",
        "Role": Object {
          "Fn::GetAtt": Array [
            "TodoLambdaServiceRoleB5E6BE45",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "TodoLambdaServiceRoleB5E6BE45": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "TodoLambdaServiceRoleDefaultPolicy7E5F178B": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "dynamodb:ListStreams",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "TodoTable20200118D40ADDDF",
                        "Arn",
                      ],
                    },
                    "/stream/*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "dynamodb:DescribeStream",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "TodoTable20200118D40ADDDF",
                  "StreamArn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TodoLambdaServiceRoleDefaultPolicy7E5F178B",
        "Roles": Array [
          Object {
            "Ref": "TodoLambdaServiceRoleB5E6BE45",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TodoTable20200118D40ADDDF": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "id",
            "AttributeType": "S",
          },
        ],
        "BillingMode": "PAY_PER_REQUEST",
        "KeySchema": Array [
          Object {
            "AttributeName": "id",
            "KeyType": "HASH",
          },
        ],
        "StreamSpecification": Object {
          "StreamViewType": "NEW_IMAGE",
        },
        "TableName": "TodoTable-20200118",
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Delete",
    },
    "TodoTableToLogInputLambdaEventSourcemapping88EDD31F": Object {
      "Properties": Object {
        "BatchSize": 3,
        "EventSourceArn": Object {
          "Fn::GetAtt": Array [
            "TodoTable20200118D40ADDDF",
            "StreamArn",
          ],
        },
        "FunctionName": Object {
          "Ref": "LogInputLambda9AB18616",
        },
        "MaximumBatchingWindowInSeconds": 10,
        "MaximumRetryAttempts": 3,
        "StartingPosition": "LATEST",
      },
      "Type": "AWS::Lambda::EventSourceMapping",
    },
    "TodoTableToTodoLambdaEventSourcemappingA0C7552A": Object {
      "Properties": Object {
        "BatchSize": 3,
        "EventSourceArn": Object {
          "Fn::GetAtt": Array [
            "TodoTable20200118D40ADDDF",
            "StreamArn",
          ],
        },
        "FunctionName": Object {
          "Ref": "TodoLambda824170AD",
        },
        "MaximumBatchingWindowInSeconds": 10,
        "MaximumRetryAttempts": 3,
        "StartingPosition": "LATEST",
      },
      "Type": "AWS::Lambda::EventSourceMapping",
    },
  },
}
`;
